gpii/node_modules/flowManager/src/FlowManagerUtilities.js

```diff
@@ -43,35 +42,33 @@
         });
     };
 
+    gpii.flowManager.loadSolutionsRegistry = function (flowManager) {
+        var resolvedFolder = fluid.module.resolvePath(flowManager.options.solutionsRegistrySourceDir);
+        flowManager.options.fullSolutionsRegistry = require(resolvedFolder);
+    };
+
     /*
      * Asynchronous function which makes a get call to the solutions registry (1st parameter) to
      * retrieve the solutions registry matching what is passed in the `device` parameter.
      * This is appended to the matchmaker payload (mmpayload) parameter, which in turn is passed
      * as parameter in the event fired.
      *
-     * @solutionsRegistry (Object) - a solutions registry data source
-     * @mmpayload (Object) - the matchmaker payload on which to attach the retrieved solution object
-     * @device (Object) - output from a device reporter. Used to filter solutions registry entries
+     * @fullSolutionsRegistry (Object) - a folder containing the full solutions registry
```

Comment:
Bad name for this option.
Also the fact that it is a physical directory has been broadcast throughout the system. Just because it's implemented as a set of files doesn't mean that it shouldn't be a component. It should be returned to being a component with a reasonable API so that it is ready to be converted back into a dataSource again one day.